bccsp/sw/fileks.go

```diff
@@ -355,27 +359,51 @@ func (ks *fileBasedKeyStore) loadKey(alias string) ([]byte, error) {
 func (ks *fileBasedKeyStore) createKeyStoreIfNotExists() error {
 	// Check keystore directory
 	ksPath := ks.path
-	missing, err := utils.DirMissingOrEmpty(ksPath)
 
-	if missing {
-		logger.Debugf("KeyStore path [%s] missing [%t]: [%s]", ksPath, missing, utils.ErrToString(err))
+	// Check if keystore directory exists
+	_, err := os.Stat(ksPath)
+	if os.IsNotExist(err) {
+		logger.Debugf("KeyStore path [%s] does not exist", ksPath)
 
-		err := ks.createKeyStore()
+		err = ks.createKeyStore()
 		if err != nil {
 			logger.Errorf("Failed creating KeyStore At [%s]: [%s]", ksPath, err.Error())
 			return nil
 		}
+	} else if err != nil {
+		return err
 	}
 
-	return nil
+	// Check if keystore directory is empty
+	f, err := os.Open(ksPath)
+	if err != nil && err != io.EOF {
+		return err
+	}
+	defer f.Close()
+
+	_, err = f.Readdir(1)
+	if err == io.EOF {
```

Comment:
```suggestion
	if err != nil && err != io.EOF {
	    return err
	 }
```
This ensures consistency within the function (similar to the one at line 379).